torch/_decomp/decompositions.py

```diff
@@ -309,6 +453,18 @@ def binary_cross_entropy_backward(
     return result
 
 
+@register_decomposition(aten._euclidean_dist)
+def _euclidean_dist(x1: Tensor, x2: Tensor) -> Tensor:
+    x1_norm = x1.pow(2).sum(-1, True)
+    x1_pad = torch.ones_like(x1_norm, memory_format=torch.contiguous_format)
```

Comment:
This makes me wonder if we should be checking that the reference implementation matches memory layout of the real one :/